### PR TITLE
fix: batch transform update for sagemaker reranker integration

### PR DESCRIPTION
**Pull request by zac-li**
**Goals:**
This PR made a required improvement (based off of https://github.com/jina-ai/jina/pull/6055/) on the SageMaker Batch Transform routine so that Reranking SageMaker model can perform Batch Transform as expected.

Previously, Jina Batch Transform endpoint has no problem reading simple data in each row from CSV and construct a input model and perform inference, for example, for 

`0f37fc919deb46e8bcafd92f1feba65f, simple text`, it will be converted to:

```python
TextDoc(
    id='0f37fc919deb46e8bcafd92f1feba65f',
    text='simple text'
)
```

However, for more complex model (e.g., includes `List` of models), such as:

```python
class RankInput(OriginalRankInput):
    query: Union[str, TextDoc]
    documents: List[TextDoc]
    top_n: Optional[int]
```
which is represented as such in CSV:

```
07937cd63aed436b9e6c34a86783f55f,where's the dog,[{\"text\": \"the dog is in my house\"}\, {\"text\": \"the cat looks good\"}\, {\"text\": \"fish chips\"}],2
```
 The original implementation would fail when parsing.

So this PR proposed a `construct_model_from_line` that can deal with more complex cases like above.

Test cases have been added to verify the fix.

### CODE REVIEW

File: tests/integration/docarray_v2/sagemaker/test_reranking.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    def __enter__(self):
+        self._old_cwd.append(os.getcwd())
+        os.chdir(self.path)
+
+    def __exit__(self, *excinfo):
+        os.chdir(self._old_cwd.pop())
+
+
+def test_provider_sagemaker_pod_inference():
+    with chdir(os.path.join(os.path.dirname(__file__), "SampleRerankerExecutor")):
```
Comment (JoanFM) [inline tests/integration/docarray_v2/sagemaker/test_reranking.py:29]: u do not need this, u can pass the path to the `config.yml` without this

File: tests/integration/docarray_v2/sagemaker/test_reranking.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            assert resp.status_code == 200
+            resp_json = resp.json()
+            assert len(resp_json["data"]) == 1
+            assert (
+                resp_json["data"][0]["results"][0]["document"]["text"] == "first result"
+            )
+
+
+def test_provider_sagemaker_pod_batch_transform_for_reranker_valid():
+    with chdir(os.path.join(os.path.dirname(__file__), "SampleRerankerExecutor")):
```
Comment (JoanFM) [inline tests/integration/docarray_v2/sagemaker/test_reranking.py:69]: same here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/runtimes/worker/http_sagemaker_app.py [hunk 3]
```diff
@@ -145,6 +154,47 @@ async def post(request: Request):
...
                     )
 
+                def construct_model_from_line(
+                    model: Type[BaseModel], line: List[str]
+                ) -> BaseModel:
+                    parsed_fields = {}
+                    model_fields = model.__fields__
+
+                    for field_name, field_str, field_info in zip(
+                        model_fields.keys(), line, model_fields.values()
```
Comment (JoanFM) [inline jina/serve/runtimes/worker/http_sagemaker_app.py:164]: this looks like a very weird way to iterate.

```suggestion
                        for field_str, (field_name, field_info) in zip (line, model_fields.items())
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/serve/runtimes/worker/http_sagemaker_app.py (hunk 3 above)
Comment (zac-li) [reply jina/serve/runtimes/worker/http_sagemaker_app.py:164]: ah right

File: tests/integration/docarray_v2/sagemaker/test_reranking.py (hunk 2 above)
Comment (zac-li) [reply tests/integration/docarray_v2/sagemaker/test_reranking.py]: fixed

File: tests/integration/docarray_v2/sagemaker/test_reranking.py (hunk 1 above)
Comment (zac-li) [reply tests/integration/docarray_v2/sagemaker/test_reranking.py]: fixed

**Pull request merged**